# Abhishek Sharma

### CS 2nd Year | Section : "I" | Roll no.: 01

### Enrollment No.: 12019009001127

## Advanced Artificial Intelligence and Machine Learning Lab 3

#### Date : 23.02.2021 | Faculty : PKB, MB, BM

In [1]:
from sklearn.datasets import load_boston

In [2]:
boston = load_boston()

In [3]:
import pandas as pd

In [4]:
data = pd.DataFrame(boston.data, columns = boston.feature_names)

In [5]:
data['MEDV'] = pd.DataFrame(boston.target) 

In [6]:
pd.DataFrame (data.corr().round(2))

CRIM    ZN  INDUS  CHAS   NOX    RM   AGE   DIS   RAD   TAX  PTRATIO  \
CRIM     1.00 -0.20   0.41 -0.06  0.42 -0.22  0.35 -0.38  0.63  0.58     0.29   
ZN      -0.20  1.00  -0.53 -0.04 -0.52  0.31 -0.57  0.66 -0.31 -0.31    -0.39   
INDUS    0.41 -0.53   1.00  0.06  0.76 -0.39  0.64 -0.71  0.60  0.72     0.38   
CHAS    -0.06 -0.04   0.06  1.00  0.09  0.09  0.09 -0.10 -0.01 -0.04    -0.12   
NOX      0.42 -0.52   0.76  0.09  1.00 -0.30  0.73 -0.77  0.61  0.67     0.19   
RM      -0.22  0.31  -0.39  0.09 -0.30  1.00 -0.24  0.21 -0.21 -0.29    -0.36   
AGE      0.35 -0.57   0.64  0.09  0.73 -0.24  1.00 -0.75  0.46  0.51     0.26   
DIS     -0.38  0.66  -0.71 -0.10 -0.77  0.21 -0.75  1.00 -0.49 -0.53    -0.23   
RAD      0.63 -0.31   0.60 -0.01  0.61 -0.21  0.46 -0.49  1.00  0.91     0.46   
TAX      0.58 -0.31   0.72 -0.04  0.67 -0.29  0.51 -0.53  0.91  1.00     0.46   
PTRATIO  0.29 -0.39   0.38 -0.12  0.19 -0.36  0.26 -0.23  0.46  0.46     1.00   
B       -0.39  0.18  -0.36  0.05 -0.38  0.13 -0.27  0.29 -0.44 -0.44    -0.18   
LSTAT    0.46 -0.41   0.60 -0.05  0.59 -0.61  0.60 -0.50  0.49  0.54     0.37   
MEDV    -0.39  0.36  -0.48  0.18 -0.43  0.70 -0.38  0.25 -0.38 -0.47    -0.51   

            B  LSTAT  MEDV  
CRIM    -0.39   0.46 -0.39  
ZN       0.18  -0.41  0.36  
INDUS   -0.36   0.60 -0.48  
CHAS     0.05  -0.05  0.18  
NOX     -0.38   0.59 -0.43  
RM       0.13  -0.61  0.70  
AGE     -0.27   0.60 -0.38  
DIS      0.29  -0.50  0.25  
RAD     -0.44   0.49 -0.38  
TAX     -0.44   0.54 -0.47  
PTRATIO -0.18   0.37 -0.51  
B        1.00  -0.37  0.33  
LSTAT   -0.37   1.00 -0.74  
MEDV     0.33  -0.74  1.00

Here we are trying to figure out the mean squared error value as low as possible using the **``` RM ```** and **``` ZN ```** value with respect to the **``` MEDV ```** values (target values)

In [8]:
x = data[['RM','ZN']]

In [9]:
y = data['MEDV']

In [10]:
from sklearn.model_selection import train_test_split

In [11]:
x_train1, x_test1, y_train1, y_test1 = train_test_split(x,y, test_size = 0.2)

In [12]:
from sklearn.linear_model import LinearRegression

In [13]:
mlr = LinearRegression()

In [14]:
mlr.fit(x_train1, y_train1)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [15]:
yTestPredict = mlr.predict(x_test1)

In [16]:
import numpy as np
from sklearn.metrics import mean_squared_error

In [17]:
np.sqrt(mean_squared_error(y_test1, yTestPredict))

5.08804702188216

### *Conclusion* : Mean Squared Error Value is ```5.09```

In [18]:
print (boston.target)

[24.  21.6 34.7 33.4 36.2 28.7 22.9 27.1 16.5 18.9 15.  18.9 21.7 20.4
 18.2 19.9 23.1 17.5 20.2 18.2 13.6 19.6 15.2 14.5 15.6 13.9 16.6 14.8
 18.4 21.  12.7 14.5 13.2 13.1 13.5 18.9 20.  21.  24.7 30.8 34.9 26.6
 25.3 24.7 21.2 19.3 20.  16.6 14.4 19.4 19.7 20.5 25.  23.4 18.9 35.4
 24.7 31.6 23.3 19.6 18.7 16.  22.2 25.  33.  23.5 19.4 22.  17.4 20.9
 24.2 21.7 22.8 23.4 24.1 21.4 20.  20.8 21.2 20.3 28.  23.9 24.8 22.9
 23.9 26.6 22.5 22.2 23.6 28.7 22.6 22.  22.9 25.  20.6 28.4 21.4 38.7
 43.8 33.2 27.5 26.5 18.6 19.3 20.1 19.5 19.5 20.4 19.8 19.4 21.7 22.8
 18.8 18.7 18.5 18.3 21.2 19.2 20.4 19.3 22.  20.3 20.5 17.3 18.8 21.4
 15.7 16.2 18.  14.3 19.2 19.6 23.  18.4 15.6 18.1 17.4 17.1 13.3 17.8
 14.  14.4 13.4 15.6 11.8 13.8 15.6 14.6 17.8 15.4 21.5 19.6 15.3 19.4
 17.  15.6 13.1 41.3 24.3 23.3 27.  50.  50.  50.  22.7 25.  50.  23.8
 23.8 22.3 17.4 19.1 23.1 23.6 22.6 29.4 23.2 24.6 29.9 37.2 39.8 36.2
 37.9 32.5 26.4 29.6 50.  32.  29.8 34.9 37.  30.5 36.4 31.1 29.1 50.
 33.3 3

In [19]:
data.head()

CRIM    ZN  INDUS  CHAS    NOX     RM   AGE     DIS  RAD    TAX  \
0  0.00632  18.0   2.31   0.0  0.538  6.575  65.2  4.0900  1.0  296.0   
1  0.02731   0.0   7.07   0.0  0.469  6.421  78.9  4.9671  2.0  242.0   
2  0.02729   0.0   7.07   0.0  0.469  7.185  61.1  4.9671  2.0  242.0   
3  0.03237   0.0   2.18   0.0  0.458  6.998  45.8  6.0622  3.0  222.0   
4  0.06905   0.0   2.18   0.0  0.458  7.147  54.2  6.0622  3.0  222.0   

   PTRATIO       B  LSTAT  MEDV  
0     15.3  396.90   4.98  24.0  
1     17.8  396.90   9.14  21.6  
2     17.8  392.83   4.03  34.7  
3     18.7  394.63   2.94  33.4  
4     18.7  396.90   5.33  36.2

Here we are trying to figure out the mean squared error value as low as possible using the **``` RM ```** , **``` ZN ```** and **```LSTAT```** value with respect to the **``` MEDV ```** values (target values)

In [20]:
x2 = data[['RM','ZN', 'LSTAT']]

In [21]:
x_train1, x_test1, y_train1, y_test1 = train_test_split(x2,y, test_size = 0.2)

In [22]:
mlr.fit(x_train1, y_train1)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [23]:
yTestPredict2 = mlr.predict(x_test1)

In [24]:
np.sqrt(mean_squared_error(y_test1, yTestPredict2))

4.651426494186151

### *Conclusion* : Mean Squared Value Error is ```4.65```

### Merging both the results in a single table : (Value of Mean Squared Error)

|Mean Squared Error depending on the Variables|Value|
|:---:|:---:|
|RM and ZN | 5.08804702188216|
|RM, ZN and LSTAT|4.65142649418615|

**********************************************************************